lib/eslint.js

```diff
@@ -302,12 +302,12 @@ function modifyConfigsFromComments(filename, ast, config, reportingConfig, messa
     // apply environment configs
     Object.keys(commentConfig.env).forEach(function (name) {
         if (environments[name]) {
-            util.mergeConfigs(commentConfig, environments[name]);
+            assign(commentConfig, environments[name]);
```

Comment:
Are you sure you want a shallow copy here? Seems like this should also be a deep merge.